symphony/bdk/core/auth/authenticator_factory.py

```diff
@@ -0,0 +1,34 @@
+from symphony.bdk.core.auth.exception.bdk_authentication_exception import AuthInitializationException
+from symphony.bdk.core.auth.bot_authenticator import BotAuthenticator, BotAuthenticatorRSA
+from symphony.bdk.core.client.api_client_factory import ApiClientFactory
+from symphony.bdk.core.config.model.bdk_config import BdkConfig
+
+
+class AuthenticatorFactory:
+    """Authenticator Factory class
+
+    Provides new instances for the main authenticators :
+    Bot Authenticator   : to authenticate the main Bot service account
+    OboAuthenticator    : to perform on-behalf-of authentication
+    """
+
+    def __init__(self, config: BdkConfig, api_client_factory: ApiClientFactory):
+        self.__config = config
+        self.__api_client_factory = api_client_factory
+
+    def get_bot_authenticator(self) -> BotAuthenticator:
+        """
+        Creates a new instance of a Bot Authenticator service.
+
+        Returns: a new BotAuthenticator instance.
+
+        """
+        if self.__config.bot.is_rsa_authentication_configured:
+            if not self.__config.bot.is_rsa_configuration_valid:
```

Comment:
I believe it is with the parentheses : `self.__config.bot.is_rsa_configuration_valid()`.